### PR TITLE
Consumption: Add Top parameter to pricesheet operation

### PR DESCRIPTION
**Pull request by bgsky**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X ] Please add REST spec PR link to the SDK PR
- [X ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X ] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@bgsky still failing

File: src/SDKs/_metadata/consumption_resource-manager.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-2018-02-03 01:21:07 UTC
+2018-05-18 01:00:05 UTC
 
 1) azure-rest-api-specs repository information
 GitHub user: Azure
 Branch:      master
-Commit:      976e0032e80d6fc286298aa3a777b3c6dc2379ca
+Commit:      cf93d01f4fe7c177b24056ff2599a164bd136d68
```
Comment (dsgouda) [inline src/SDKs/_metadata/consumption_resource-manager.txt]: Please create a generate.ps1 similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code.

**Comment by dsgouda** [comment]
Please link the REST spec PR

**Review by dsgouda — CHANGES_REQUESTED** [review]
There are some changes introduced in the code here. Please bump the version to `18.0.1`

**Comment by dsgouda** [comment]
@bgsky we do have a script [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/tools/HelperUtilities/psScripts/Copy-SessionRecords.ps1) that can help you copy the session records.
Do let us know if you have any suggestions for improvements.

**Comment by dsgouda** [comment]
@bgsky  Please create a generate.ps1 as suggested and re-generate the code

**Comment by bgsky** [comment]
@dsgouda Would you let me know if there is any difference if the code is re-generated, or generate.ps1 just introduces a new protocol

**Comment by dsgouda** [comment]
Yes, as a protocol we plan to move RPs towards using the ps1 script code instead of the cmd.
As such the code generated should not show any differences

**Review by dsgouda — APPROVED** [review]
LGTM will merge on CIs passing

**Comment by bgsky** [comment]
@dsgouda The latest iteration adds generate.ps1 file and updates metadata file on consumption. 

I have played with generate.ps1 and Copy-SessionRecords.ps1 files, while not adding all generated files into the pull request as the latest version of Consumption swagger file includes many other operations. We plan to keep the current pull request small with previous version, and update it with the latest version later. 

Copy-SessionRecords.ps1 works great. Thanks for adding this tool. It does require source and destination root directories manually added.

**Comment by dsgouda** [comment]
@bgsky The best way to select exactly which operations you would like to generate the sdk for is to filter out the files using tags in the configuration file (and set the default tag to whichever is the one you want to generate sdk for). With more improvements coming in we will allow specifying the tags in the generate.ps1 script itself
Due to some implementation limitations we have decided to keep `Copy-SessionRecords.ps1` simple for the first iteration, would certainly appreciate your feedback

**Comment by dsgouda** [comment]
Please disregard Travis failures here, working on getting it fixed now

**Comment by bgsky** [comment]
@dsgouda Just a smalll follow-up, would you have a link of tutorial for our side to publish a new version of nuget package of consumption SDK

**Comment by dsgouda** [comment]
Most of the documentation to generate and publish an sdk is available [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/Documentation/code-generation-instructions.md)

**Comment by bgsky** [comment]
Thanks, guess it has information needed.

**Pull request merged**